apl-core/src/main/java/com/apollocurrency/aplwallet/apl/core/rest/utils/ResponseBuilder.java

```diff
@@ -0,0 +1,107 @@
+/*
+ *
+ *  Copyright © 2018-2019 Apollo Foundation
+ *
+ */
+
+package com.apollocurrency.aplwallet.apl.core.rest.utils;
+
+import com.apollocurrency.aplwallet.api.dto.BaseDTO;
+import com.apollocurrency.aplwallet.api.response.ResponseBase;
+import com.apollocurrency.aplwallet.api.response.ResponseDone;
+import com.apollocurrency.aplwallet.apl.core.rest.ApiErrors;
+import com.apollocurrency.aplwallet.apl.core.rest.ErrorInfo;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import javax.ws.rs.core.Response;
+
+public class ResponseBuilder {
+
+    private static ObjectMapper mapper = new ObjectMapper();
+
+    private final long startRequestTime;
+    protected BaseDTO dto;
+    protected ResponseBase response;
+    protected int status;
+
+    protected ResponseBuilder(int status) {
+        this.status = status;
+        this.startRequestTime = System.currentTimeMillis();
+    }
+
+    public static ResponseBuilder startTiming(){
+        return new ResponseBuilder(200);
```

Comment:
Status 200 is correct only for successful requests, for errors required 4xx codes such as 422 for validation

Reply:
It's reasonable, but as I got, the 'Working UI' is the main requirement. Currently, the UI expects that when the error occurs the errorCode field will set to a non-zero value but the HTTP status will set to 200.